tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/BsonSerialization/BsonSerializationHelper.cs

```diff
@@ -167,6 +181,9 @@ private BsonHelper(
                 return null;
             }
 
+            // in v3+ IBsonWriter has extra methods, so we need to handle that differently
+            var isV3Bson = ibsonWriterType.GetMethod(nameof(IBsonWriterProxyV3.WriteGuid), [typeof(Guid)]) is not null;
```

Comment:
I wonder if a version check on the assembly might be better, but no strong feelings

Reply:
Yeah, I was torn tbh, no strong feelings either 🤷‍♂️

Reply:
As discussed... compared to:

```csharp
var isV3Bson = ibsonWriterType.Assembly.GetName().Version?.Major >= 3;
```

|       Method |      Mean |    Error |   StdDev |  Gen 0 | Allocated |
|------------- |----------:|---------:|---------:|-------:|----------:|
|    HasMethod |  15.68 ns | 0.155 ns | 0.145 ns |      - |         - |
| VersionCheck | 438.11 ns | 6.453 ns | 6.037 ns | 0.0019 |     584 B |